patterns/gitops/multi-cluster-hub-spoke-argocd/hub/main.tf

```diff
@@ -0,0 +1,297 @@
+  }
+  addons = merge(
+    local.aws_addons,
+    local.oss_addons,
+    { kubernetes_version = local.cluster_version },
+    { aws_cluster_name = module.eks.cluster_name }
+  )
+
+  addons_metadata = merge(
+    module.eks_blueprints_addons.gitops_metadata,
+    {
+      aws_cluster_name = module.eks.cluster_name
+      aws_region       = local.region
+      aws_account_id   = data.aws_caller_identity.current.account_id
+      aws_vpc_id       = module.vpc.vpc_id
+    },
+    {
+      argocd_iam_role_arn = module.argocd_irsa.iam_role_arn
+      argocd_namespace    = local.argocd_namespace
+    },
+    {
+      addons_repo_url      = local.gitops_addons_url
+      addons_repo_basepath = local.gitops_addons_basepath
+      addons_repo_path     = local.gitops_addons_path
+      addons_repo_revision = local.gitops_addons_revision
+    }
+  )
+
+  argocd_apps = {
+    addons    = file("${path.module}/bootstrap/addons.yaml")
+    workloads = file("${path.module}/bootstrap/workloads.yaml")
+  }
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/gitops-bridge-dev/gitops-bridge"
+  }
+}
+
+################################################################################
+# GitOps Bridge: Bootstrap
+################################################################################
+module "gitops_bridge_bootstrap" {
+  source = "github.com/gitops-bridge-dev/gitops-bridge-argocd-bootstrap-terraform?ref=v2.0.0"
+
+  cluster = {
+    cluster_name = module.eks.cluster_name
+    environment  = local.environment
+    metadata     = local.addons_metadata
+    addons       = local.addons
+  }
+  apps = local.argocd_apps
+  argocd = {
+    namespace = local.argocd_namespace
+  }
+}
+
+################################################################################
+# ArgoCD EKS Access
+################################################################################
+module "argocd_irsa" {
+  source = "aws-ia/eks-blueprints-addon/aws"
+
+  create_release             = false
+  create_role                = true
+  role_name_use_prefix       = false
+  role_name                  = "${module.eks.cluster_name}-argocd-hub"
+  assume_role_condition_test = "StringLike"
+  create_policy              = false
+  role_policies = {
+    ArgoCD_EKS_Policy = aws_iam_policy.irsa_policy.arn
+  }
+  oidc_providers = {
+    this = {
+      provider_arn    = module.eks.oidc_provider_arn
+      namespace       = local.argocd_namespace
+      service_account = "argocd-*"
+    }
+  }
+  tags = local.tags
+
+}
+
+resource "aws_iam_policy" "irsa_policy" {
+  name        = "${module.eks.cluster_name}-argocd-irsa"
+  description = "IAM Policy for ArgoCD Hub"
+  policy      = data.aws_iam_policy_document.irsa_policy.json
+  tags        = local.tags
+}
+
+data "aws_iam_policy_document" "irsa_policy" {
+  statement {
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["sts:AssumeRole"]
+  }
+}
```

Comment:
not a blocker but you can consider changing to https://registry.terraform.io/modules/terraform-aws-modules/iam/aws/latest/submodules/iam-role-for-service-accounts-eks

Reply:
yes it makes sense